File: minimum-cost-to-connect-sticks/minimum-cost-to-connect-sticks.cs
public class Solution {
    public int ConnectSticks(int[] sticks) {
        
        //Minheap
        
        PriorityQueue<int,int> minheap=new PriorityQueue<int,int>();
        
        for(int i=0;i<sticks.Length;i++)
        {
            minheap.Enqueue(sticks[i],sticks[i]);
        }
        
        int result=0;
        
        while(minheap.Count!=1)
        {
            int stick1=minheap.Dequeue();
            int stick2=minheap.Dequeue();
            int cost= stick1+stick2;
            result=result+cost;
            
            minheap.Enqueue(cost,cost);
        }
        
        return result;
        
    }
}

File: minimum-cost-to-connect-sticks/minimum-cost-to-connect-sticks.java
class Solution {
    public int connectSticks(int[] sticks) {
        
        PriorityQueue<Integer> queue=new PriorityQueue<>();
        
        for(int i=0;i<sticks.length;i++)
        {
            queue.add(sticks[i]);    
        }
        
        int result=0;
        
        while(queue.size()!=1)
        {
            int newNum=queue.remove()+queue.remove();
            result= result+newNum;
            queue.add(newNum);
        }
        
        return result;
    }
}